packages/studio/src/components/AddPageButton/StreamScopeCollector.tsx

```diff
@@ -1,58 +1,185 @@
     },
-    [handleConfirm, actions]
+    [streamScope, setStreamScope]
+  );
+
+  const scopeFormData: {
+    [field in keyof StreamScope]: Pick<
+      ScopeFieldProps,
+      "emptyText" | "allItems" | "getDisplayName"
+    >;
+  } = useMemo(
+    () => ({
+      entityIds: {
+        emptyText: "No entities found in the account.",
+        // TODO (SLAP-2907): Populate dropdown from store
+        allItems: [],
+      },
+      entityTypes: {
+        emptyText: "No entity types found in the account.",
+        allItems: Object.keys(entitiesRecord),
+      },
+      savedFilterIds: {
+        emptyText: "No saved filters found in the account.",
+        allItems: savedFilters.map((f) => f.id),
+        getDisplayName: (item) => {
+          const savedFilter = savedFilters.find((f) => f.id === item);
+          return savedFilter ? getDisplayString(savedFilter) : item;
+        },
+      },
+    }),
+    [savedFilters, entitiesRecord]
   );
 
+  const modalBodyContent = useMemo(() => {
+    const totalStreamScopeItems = Object.values(streamScope ?? {}).reduce(
+      (numItems, scopeItems) => {
+        return numItems + scopeItems.length;
+      },
+      0
+    );
+
+    return (
+      <>
+        <div className="italic mb-4">
+          Use one of the optional fields below to specify which entities this
+          page can access.
+        </div>
+        {Object.entries(scopeFormData).map(([field, data]) => {
+          const selectedItems: string[] | undefined = streamScope?.[field];
+          const hasOtherScopeFilters =
+            totalStreamScopeItems > (selectedItems?.length ?? 0);
+          return (
+            <ScopeField
+              key={field}
+              field={field}
+              selectedItems={selectedItems}
+              updateScopeItems={updateScopeItems(field)}
+              disabled={data.allItems.length > 0 && hasOtherScopeFilters}
+              {...streamScopeFormData[field]}
+              {...data}
+            />
+          );
+        })}
+      </>
+    );
+  }, [scopeFormData, streamScope, updateScopeItems]);
+
   return (
-    <FormModal
+    <DialogModal
       isOpen={isOpen}
       title="Content Scope"
-      instructions="Use the optional fields below to specify which entities this page can access. Values should be separated by commas."
-      formData={streamScopeFormData}
-      closeOnConfirm={false}
-      confirmButtonText="Next"
       handleClose={handleClose}
-      handleConfirm={onConfirm}
+      handleConfirm={handleConfirm}
+      body={modalBodyContent}
+      confirmButtonText="Next"
+      isConfirmButtonDisabled={false}
     />
   );
 }
+
+interface ScopeFieldProps {
+  field: string;
+  description: string;
+  allItems: string[];
+  tooltip: string;
+  emptyText: string;
+  selectedItems: string[] | undefined;
+  updateScopeItems: (selectedItems: string[] | undefined) => void;
+  getDisplayName?: (item: string) => string;
+  disabled?: boolean;
+}
+
+function ScopeField({
```

Comment:
Could we put this in its own file, maybe in `src/components/common`? We'll want to reuse this when we update the Edit Flow.